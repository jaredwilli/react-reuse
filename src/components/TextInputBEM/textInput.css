/* TextInput using BEM styling principles */

.textinput {
    margin-bottom: 16px;
}

.textinput__error {
    color: red;
}

.textinput__input--state-error {
    border: 1px solid red;
}
