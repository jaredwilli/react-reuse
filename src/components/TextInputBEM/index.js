import './textInput.css';
export { default } from './TextInputBEM';
