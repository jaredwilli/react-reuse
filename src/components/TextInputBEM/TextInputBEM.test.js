import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TextInputBEM from './TextInputBEM';

describe('TextInputBEM', () => {
    it('renders the component correctly', () => {
        const tree = renderer.create(<TextInputBEM />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
