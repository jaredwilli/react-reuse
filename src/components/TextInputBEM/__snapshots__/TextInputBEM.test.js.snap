// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputBEM renders the component correctly 1`] = `
<div
  className="text-input"
>
  <label
    htmlFor={undefined}
    style={
      Object {
        "display": "block",
      }
    }
  >
     
  </label>
  <input
    className={undefined}
    id={undefined}
    name={undefined}
    onChange={undefined}
    placeholder={undefined}
    type="text"
    value={undefined}
  />
</div>
`;
